sensor/kubernetes/listener/resources/rbac/binding_fetcher.go

```diff
@@ -49,18 +59,34 @@ func (r *bindingFetcher) generateManyDependentEvents(bindings []namespacedBindin
 // role bindings were updated by re-sync. The update events on RoleBindings would pick up any created/removed roles and update
 // RoleID accordingly.
 func (r *bindingFetcher) generateDependentEvent(relatedBinding namespacedBindingID, updateRoleID string, isClusterRole bool) (*central.SensorEvent, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 	if relatedBinding.IsClusterBinding() {
-		clusterBinding, err := r.k8sAPI.RbacV1().ClusterRoleBindings().Get(context.TODO(), relatedBinding.name, metav1.GetOptions{})
-		if err != nil {
-			return nil, errors.Wrapf(err, "fetching k8s API for ClusterRoleBinding %s", relatedBinding.name)
+		var clusterRoleBinding *v1.ClusterRoleBinding
+		if err := retry.WithRetry(func() error {
```

Comment:
Why retrying here and not in the caller `generateManyDependentEvents`
The retry function is the same and there we need to do it once

Reply:
I've wrapped both calls into a single retry function but I kept it in `generateDependentEvent`. I don't know why but it looks cleaner to me plus no need to updated all the tests 🙂 Let me know what you think.